### PR TITLE
DOC Update references to CMS 3 and 4

### PR DESCRIPTION
**Pull request by GuySartorelli**
For CMS5 docs, we don't care about CMS3. We also want anything saying that 4 is the latest to instead refer to 5.
This PR also removes some references to specific CMS major versions because the language doesn't need to be that specific. This means future major versions won't need to update those same references.

## Parent issue:
- https://github.com/silverstripe/developer-docs/issues/44

### CODE REVIEW

File: en/02_Developer_Guides/09_Security/04_Secure_Coding.md [hunk 1]
```diff
@@ -710,7 +710,7 @@ following in your .htaccess to ensure this behaviour is activated.
 </IfModule>
 ```
 
-As of Silverstripe CMS 4, this behaviour is on by default, and the environment variable is no longer required. For correct operation, it is necessary to always set `SS_TRUSTED_PROXY_IPS` if using a proxy.
+This behaviour is on by default; the environment variable is not required. For correct operation, it is necessary to always set `SS_TRUSTED_PROXY_IPS` if using a proxy.
```
Comment (sabina-talipova) [inline en/02_Developer_Guides/09_Security/04_Secure_Coding.md:713]: Is this correct to have a semicolon: 
`... behaviour is on by default ;  ...`

**Comment by sabina-talipova** [comment]
Just tiny question. You removed almost all sections in `03_Upgrading` folder. Don't we need them anymore?

File: en/02_Developer_Guides/09_Security/04_Secure_Coding.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/09_Security/04_Secure_Coding.md:713]: Yup.
https://www.merriam-webster.com/words-at-play/a-guide-to-using-semicolons
> A semicolon separates related independent clauses that are joined without a coordinating conjunction, such as _and_:
> > Mabel the Cat had made a big pot of goulash; it was simmering on the stove.

**Comment by GuySartorelli** [comment]
> You removed almost all sections in 03_Upgrading folder. Don't we need them anymore?

- `02_Upgrading_module.md` had explicit advice for upgrading modules from v3 to v4. It's possible we'll have something like this document for CMS5, but at this stage it's cleaner to remove it, and then create a new one if we have something that is relevant explicitly for modules.
- `03_Migrating_off_CWP.md` was relevant when the CWP recipes were replaced with ccl recipes. Anyone upgrading to CMS 5 will almost certainly have already migrated off the CWP recipes.
- `04_Upgrading_project.md` has already been replaced with [`02_Upgrading_your_project.md`](https://github.com/silverstripe/developer-docs/blob/5/en/03_Upgrading/02_Upgrading_your_project.md).
- `06_Upgrading_to_GraphQL_4.md` is probably still relevant. I'll update that one to be CMS5 centric instead of removing it.

**Pull request merged**